configure.py

```diff
@@ -1,154 +1,188 @@
+n.rule("ld", command="$compiler $in $ldflags -o $out", description="LD $out")
 n.newline()
+n.rule("ld_shared", command="$compiler -shared $in $ldflags -o $out", description="LD_SHARED $out")
+n.newline()
+
+# Download dependancies
+deps_path = os.path.join(build_dir, "_deps")
+nanobind_path = os.path.join(deps_path, "nanobind")
+robin_path = os.path.join(deps_path, "robin-map")
 
+os.makedirs(deps_path, exist_ok=True)
+
+if not os.path.isdir(nanobind_path):
+    subprocess.run(["git", "clone", "https://github.com/wjakob/nanobind", nanobind_path], check=True)
+
+if not os.path.isdir(robin_path):
+    subprocess.run(["git", "clone", "https://github.com/Tessil/robin-map.git", robin_path], check=True)
+
+python_include_path = sysconfig.get_paths()["include"]
+python_lib_path = os.path.join(sysconfig.get_paths()["data"], "libs", f"libpython{sys.version_info.major}.{sys.version_info.minor}.a")
+nanobind_include_path = os.path.join(nanobind_path, "include")
+nanobind_src_path = os.path.join(nanobind_path, "src")
+robin_include_path = os.path.join(robin_path, "include")
+
+# Flags
 common_ccflags = [
-    '-DREVOLUTION',
-    '-fno-asynchronous-unwind-tables',
-    '-fno-exceptions',
-    '-fno-rtti',
-    '-fshort-wchar',
-    '-fstack-protector-strong',
-    '-isystem', '.',
-    '-isystem', 'include',
-    '-isystem', 'source',
-    '-isystem', 'vendor',
-    '-isystem', 'build',
-    '-std=c++23',
-    '-Wall',
-    '-Wdouble-promotion',
-    '-Werror',
-    '-Wextra',
-    '-Wno-delete-non-virtual-dtor',
-    '-Wno-packed-bitfield-compat',
-    '-Wsuggest-override',
+    "-DREVOLUTION",
+    "-fno-asynchronous-unwind-tables",
+    "-fno-exceptions",
+    "-fshort-wchar",
+    "-fstack-protector-strong",
+    "-isystem", "include",
+    "-isystem", "source",
+    "-Wall",
+    "-Werror",
+    "-Wextra",
+    "-Wno-delete-non-virtual-dtor",
+    "-Wno-packed-bitfield-compat",
+    "-Wsuggest-override",
+    "-std=c++23",
+    "-fPIC",
+    "-fno-exceptions",
 ]
 
-target_cflags = [
-    '-O3',
+bindings_ccflags = [
+    "-DREVOLUTION",
+    "-fstack-protector-strong",
+    "-isystem", "include",
+    "-isystem", "source",
+    "-isystem", python_include_path,
+    "-isystem", nanobind_include_path,
+    "-isystem", robin_include_path,
+    "-Wall",
+    "-Werror",
+    "-Wextra",
+    "-Wno-format",
+    "-Wno-delete-non-virtual-dtor",
+    "-Wno-packed-bitfield-compat",
+    "-Wsuggest-override",
+    "-std=c++23",
+    "-fPIC",
 ]
 
-debug_cflags = [
-    '-DBUILD_DEBUG',
-    '-O0',
-    '-ggdb',
+# Define build types
+@dataclass
+class Build:
+    name: str
+    suffix: str
+    flags: list[str]
+    obj_files: dict = field(default_factory=lambda: {
+        "common": [],
+        "main": [],
+        "bindings": []
+    })
+
+builds = [
+    Build(name="target", suffix="", flags=["-O3"]),
+    Build(name="debug", suffix="D", flags=["-DBUILD_DEBUG", "-O0", "-ggdb"])
 ]
 
-common_ldflags = []
+# Compile files
+def GetFiles():
```

Comment:
Since this is python, this should be snake case I think. Or is that just for methods?